File: queues/redis/Python/tests/test_redisclient.py
import sys
from pathlib import Path
from unittest.mock import MagicMock, patch

import pytest

sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
import redisclient


@pytest.fixture(autouse=True)
def env(monkeypatch):
    monkeypatch.setenv("REDIS_HOST", "redis.local")


def test_init_missing(monkeypatch):
    monkeypatch.delenv("REDIS_HOST", raising=False)
    monkeypatch.delenv("REDIS_URL", raising=False)
    assert redisclient.Init()["status_code"] == 601


@patch("redisclient._client")
def test_push(mock_client_fn):
    c = MagicMock()
    mock_client_fn.return_value = c
    assert redisclient.Push("q", "v") == {"status_code": 200}

File: queues/redis/C/tests/test_redisclient.c
#include "../../../../tests/c/testutil.h"
#include "../include/redisclient.h"
#include <stdlib.h>

TEST(test_init_missing)
{
    unsetenv_safe("REDIS_URL");
    unsetenv_safe("REDIS_HOST");
    char *json = redis_init();
    ASSERT("601", json_status(json) == 601);
    free(json);
}

TEST(test_init_ok)
{
    setenv("REDIS_HOST", "redis.local", 1);
    char *json = redis_init();
    ASSERT("200", json_status(json) == 200);
    free(json);
}

int main(void)
{
    RUN(test_init_missing);
    RUN(test_init_ok);
    printf("redis C tests passed\n");
    return 0;
}
